docs/design/storagenode-graceful-exit.md

```diff
@@ -0,0 +1,89 @@
+# Storage Node Graceful Exit Design Document
+
+
+## Overview
+
+When a Storage Node wants to leave the network but does not want to lose their escrow we need to have a mechanism for them to exit the network “gracefully”.
+
+This process including the Storage nodes transferring their pieces to other nodes so that the satellite does not have to repair those pieces because of a node exiting abruptly. The process of a Storage Node exiting gracefully including that node requesting a list of Storage Nodes to send their pieces to and updating the satellite with what nodes are now storing those pieces. Graceful Exit for Storage Nodes is beneficially for both the Storage Node and satellite because the storage node receives their escrow and the satellite saves money from not having to repair files.
+
+
+## Goals
+
+- Give Storage Nodes a mechanism to leave the network while receiving their escrows, and reduce repair caused by node churn on satellites. 
+
+
+## Non Goals
+
+- Sending the storage nodes escrows.
+	- The sending of tokens will happen through our normal token payment process.
+
+
+## Scenarios (MVP)
+
+- A storage node gracefully exits a satellite or all satellites.
+- A storage node fails to complete the graceful exit process.
+
+
+## Scenarios (Non MVP)
+
+- A storage node runs out of bandwidth during the graceful exit process.
+- A storage node wants to “partially” exit a satellite.
+- A storage node wants to gracefully exit one satellite.
+- A storage node wants to rejoin a satellite it previously exited.
+
+## Business Requirements/ Job Stories (MVP)
```

Comment:
Let's add these:
- Once a storage node triggers the graceful exits, the process cannot be canceled. The storage node must lead it to successful completion.
- (maybe this is Non MVP) When the graceful exit process finishes, the satellite will send a message to the storage node for the successful completion, so the storage node can remove any remaining garbage for this satellite.

Reply:
I added your first bullet point

we have this requirement which covers your second bullet point i think
- When a Storage Node gracefully exits the network I want all of the pieces they have been storing to be deleted from their hard drive so that their computers hard drive space is no longer used.